<?xml version="1.0"?>

<launch>
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="cf_list" default="['cf1']" />
  <arg name="to_sim" default="False" />

  <!-- Launch server -->
  <group unless="$(arg to_sim)">
    <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
    </include>
  </group>


  <!-- Initiate joystick -->
  <node name="joy" pkg="joy" type="joy_node" output="screen" >
    <param name="dev" value="$(arg joy_dev)" />
  </node>

  <!-- Map controller to services -->
  <node name="joystick_controller" pkg="swarm_manager" type="joy_controller.py" output="screen">
    <param name="joy_topic" value="/joy" />
    <param name="to_sim" value="$(arg to_sim)"/>

    <!-- Axis number, start at 0 -->
    <param name="x_axis" value="1"/>
    <param name="y_axis" value="0"/>
    <param name="z_axis" value="5"/>
    <param name="yaw_axis" value="2"/>
    <param name="x_velocity_max" value="30"/>
    <param name="y_velocity_max" value="-30"/>
    <param name="z_velocity_max" value="60000"/>
    <param name="yaw_velocity_max" value="-200"/>
  </node>

  <!-- Launch swarm manager-->
  <node name="swarm_manager" pkg="swarm_manager" type="swarm_manager.py" output="screen">
    <rosparam param="cf_list" subst_value="True">$(arg cf_list) </rosparam>
    <param name="to_sim" value="$(arg to_sim)"/>
  </node>

  <!-- Launch formation manager-->
  <node name="formation_manager" pkg="formation_manager" type="formation_manager.py" output="screen">
    <rosparam param="cf_list" subst_value="True">$(arg cf_list) </rosparam>
    <param name="to_sim" value="$(arg to_sim)"/>
  </node>

  <!-- Launch trajectory planner -->
  <node name="trajectory_planner" pkg="trajectory_planner" type="trajectory_planner_ros.py" output="screen">
    <rosparam param="cf_list" subst_value="True">$(arg cf_list) </rosparam>
  </node>

  <!-- Rviz markers -->
  <node name="rviz_markers" pkg="swarm_manager" type="rviz_markers.py" output="screen" >
    <rosparam param="cf_list" subst_value="True">$(arg cf_list) </rosparam>
  </node>

  <!-- run rviz -->
  <param name="cf_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find swarm_manager)/launch/crazyflie_pos.rviz" required="true" />

</launch>
